src/main/kotlin/com/cosmotech/api/utils/SecurityUtils.kt

```diff
@@ -23,17 +24,31 @@ fun getCurrentAuthenticatedIssuer(): String {
     throw IllegalStateException("User Authentication not found in Security Context")
   }
 
-  val authentication = getCurrentAuthentication() as BearerTokenAuthentication
-  return authentication.token.tokenValue.let { JWTParser.parse(it).jwtClaimsSet.issuer }
+  val authentication = getCurrentAuthentication()
+
+  if (authentication is JwtAuthenticationToken) {
+    return authentication.token.tokenValue.let { JWTParser.parse(it).jwtClaimsSet.issuer }
+  }
+
+  return (authentication as BearerTokenAuthentication).token.tokenValue.let {
+    JWTParser.parse(it).jwtClaimsSet.issuer
+  }
 }
 
 fun getCurrentAuthenticatedMail(configuration: CsmPlatformProperties): String {
   if (getCurrentAuthentication() == null) {
     throw IllegalStateException("User Authentication not found in Security Context")
   }
 
-  val authentication = getCurrentAuthentication() as BearerTokenAuthentication
-  return authentication.token.tokenValue.let {
+  val authentication = getCurrentAuthentication()
+
+  if (authentication is JwtAuthenticationToken) {
+    return authentication.token.tokenValue.let {
+      JWTParser.parse(it).jwtClaimsSet.getStringClaim(configuration.authorization.mailJwtClaim)
```

Comment:
This line is present 2 times : 
` JWTParser.parse(it).jwtClaimsSet.getStringClaim(configuration.authorization.mailJwtClaim)`
To refactor? Maybe with `when `